cgfs.py

```diff
@@ -160,34 +273,46 @@ def truncate(self, length):
         self.dirty = True
 
     def write(self, data, offset):
-        if offset > len(self.data):
-            self.data += bytes(offset - len(self.data))
-        if len(self.data) - offset - len(data) > 0:
+        if offset > self.stat['st_size']:
+            self._data += bytes(offset - len(self.data))
+
+        if self.stat['st_size'] - offset - len(data) > 0:
+            # Write in between
             second_offset = offset + len(data)
-            self.data = self.data[:offset] + data + self.data[second_offset:]
+            old_d = self.data
+            self._data = old_d[:offset] + data + old_d[second_offset:]
+        elif offset == 0:
+            self._data = data
         else:
-            self.data = self.data[:offset] + data
-        self.stat['st_size'] = len(self.data)
+            self._data = self.data[:offset] + data
+
+        self.stat['st_size'] = len(self._data)
         self.stat['st_atime'] = time()
         self.stat['st_mtime'] = time()
         self.dirty = True
         return len(data)
 
 
 class CGFS(LoggingMixIn, Operations):
-    def __init__(self, latest_only):
+    def __init__(self, latest_only, fixed=False, tmpdir=None):
         self.latest_only = latest_only
+        self.fixed = fixed
         self.files = {}
         self.fd = 0
+        self._open_files = {}
+
+        self._tmpdir = tmpdir
 
         self.files = Directory(
             {
                 'id': None,
                 'name': 'root'
             }, type=DirTypes.FSROOT
         )
+
         self.files.getattr()
         self.load_courses()
+        print('Mounted')
```

Comment:
This should only be printed in `--verbose` mode in my opinion. I would probably run the filesystem in the background (with `cgfs user mountpoint &`) and it's very annoying when backgrounded programs print to stdout.

Reply:
No this should only not be printed in `--daemon` mode imo.